packages/components/app/styles/components/application-state.scss

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// APPLICATION-STATE COMPONENT
+//
+
+// notes:
+// - we need to be able to line things up in two directions so grid was the better choice here; initially tried to imitate the modal header but the design is different
+// - constraint: don't use column-gap
+
+$hds-application-state-padding: 12px 0;
+
+.hds-application-state {
+  width: 19.5rem;
+  max-width: 100%;
+  margin: 0 auto; // this will center the component in the parent container
+}
+
+.hds-application-state__header {
+  display: grid;
+  grid-template-columns: min-content 1fr;
+  // align-items: center; // this works if the text is only one line but not multi-line
+  align-items: start;
+  color: var(--token-color-foreground-primary);
+}
+
+.hds-application-state__icon {
+  width: 32px; // 24px for icon, 8px for gap to title; normally we'd use column-gap but we have been trying to avoid that so far
+  padding-top: 4px; // this seems to align the icon along with line 21
+}
+
+.hds-application-state__title,
+.hds-application-state__error-code {
+  grid-column-start: 2;
+}
+
+.hds-application-state__body {
+  display: flex;
+  padding: $hds-application-state-padding;
+  color: var(--token-color-foreground-faint);
+}
+
+.hds-application-state__footer {
+  display: flex;
+  justify-content: space-between;
+}
+
+.hds-application-state__footer--has-divider {
+  padding-top: 16px; // waiting to hear back on design if this should really be different
```

Comment:
@andgen404 let me know when you confirm if this should be 16 or 12 like the others.

Reply:
@MelSumner Confirmed, we are now going with 12px